### ISSUE TITLE
Reduce `search_index.js` file size ?

### ISSUE THREAD

**Issue by t-bltg** [JuliaDocs/Documenter.jl#2606]
How can I reduce `search_index.js` file size ?

The `gh-pages` limit is 100Mb, and I'm hitting this limit e.g. https://github.com/JuliaPlots/Plots.jl/actions/runs/11794903812/job/32859378612#step:5:27591.

**Comment by mortenpi** [comment]
Any ideas why it's that large? The Julia manual's index is the largest I know of and it's less than 1 MiB. I suspect some sort of auto-generated content is getting included..?

**Comment by t-bltg** [comment]
There definitively is some auto-generated content.

Here is the dev version, around **99Mb**: https://github.com/JuliaPlots/PlotDocs.jl/blob/gh-pages/dev/search_index.js.

The previous stable version for `Plots@1.39.0` (quite old build) was only around **8.65Mb** https://github.com/JuliaPlots/PlotDocs.jl/blob/gh-pages/v1.39.0/search_index.js.

This coincides with the breaking release of Documenter 1.0 if I recall.

**Comment by mortenpi** [comment]
Quickly looking through the 99 MiB one, I see heaps of SVG data in there. Documenter shouldn't be adding that, so I think it's something specific to the Plots repo that is adding it. I'd suggest trying to reduce this to an MWE, to see how it's ending up in the index.

**Comment by fredrikekre** [comment]
As a sidenote, perhaps this file can be gzipped? docs.julialang.org goes from 4.1M to 950K (and the plots one from 99M to 16M).

**Comment by t-bltg** [comment]
> Quickly looking through the 99 MiB one, I see heaps of SVG data in there.

Yeah, it seems that the image such as the one in [this exemple](https://docs.juliaplots.org/dev/gallery/gr/generated/gr-ref038/) lands in the `search_index` when e.g. searching for the keyword `show_empty_bins`.
I'm wondering what can be done to avoid images inclusion.

> As a sidenote, perhaps this file can be gzipped?

Good idea !

**Comment by t-bltg** [comment]
I've been able to reduce the file size to ~19Mb with our updated fix https://github.com/JuliaPlots/Plots.jl/blob/v2/docs/make.jl#L35-L68, which was out of sync with `Documenter`'s code.

Still, being able to remove the images from the search index might be our best bet.

**Comment by fingolfin** [comment]
@t-bltg Off-topic, but: It would be interesting to know why you need to monkey patch documenter, and if we can perhaps find a way to accommodate your needs in a way that would allow you to drop that; I expect that'll be better for everyone involved on the long run.
